bin/genesis

```diff
@@ -4069,12 +4191,17 @@ sub active_certificates {
 sub target_vault {
 	my ($target) = @_;
 	if ($target) {
-		system(qw(safe target), $target) == 0 or exit 1;
+		Genesis::ShellExec::interact({passfail => 1}, 'safe target "$1"', $target) or exit 1;
 	} else {
-		system(qw(safe target -i)) == 0 or exit 1;
+		Genesis::ShellExec::interact({passfail => 1}, 'safe target -i') or exit 1;
 	}
-	chomp($ENV{GENESIS_TARGET_VAULT}=qx(safe target 2>&1 | grep '.*targeting .* at.*' | sed -e 's/.*targeting \\([^ ]*\\) at.*/\\1/'));
-	die "Could not set target vault" if $? >> 8;
+	chomp(
+		$ENV{GENESIS_TARGET_VAULT}=Genesis::ShellExec::get(
+			{onfailure => "Could not set target vault"} ,
+			'safe target 2>&1 | grep "$1" | sed -e "$2"',
+			'.*targeting .* at.*', 's/.*targeting \([^ ]*\) at.*/\1/'
+		)
+	);
```

Comment:
Put a FIXME comment in here.

safe now supports `safe target --json`, and once that gets released, we should totes use that.